src/pykeen/nn/node_piece/representations.py

```diff
@@ -0,0 +1,300 @@
+
+        :param tokenizer:
+            the tokenizer instance.
+        :param num_tokens:
+            the number of tokens to select for each entity.
+        :param token_representation:
+            the pre-instantiated token representations, or an EmbeddingSpecification to create them
+        :param token_representation_kwargs:
+            additional keyword-based parameters
+        :param mapped_triples:
+            the ID-based triples
+        :param num_entities:
+            the number of entities
+        :param num_relations:
+            the number of relations
+        :param kwargs:
+            additional keyword-based parameters passed to TokenizationRepresentation.__init__
+        :return:
+            A tokenization representation by applying the tokenizer
+        """
+        # apply tokenizer
+        vocabulary_size, assignment = tokenizer(
+            mapped_triples=mapped_triples,
+            num_tokens=num_tokens,
+            num_entities=num_entities,
+            num_relations=num_relations,
+        )
+        return TokenizationRepresentation(
+            assignment=assignment,
+            token_representation=token_representation,
+            token_representation_kwargs=token_representation_kwargs,
+            **kwargs,
+        )
+
+    def extra_repr(self) -> str:  # noqa: D102
+        return "\n".join(
+            (
+                f"max_id={self.assignment.shape[0]},",
+                f"num_tokens={self.assignment.shape[1]},",
+                f"vocabulary_size={self.vocabulary_size},",
+            )
+        )
+
+    def _plain_forward(
+        self,
+        indices: Optional[torch.LongTensor] = None,
+    ) -> torch.FloatTensor:  # noqa: D102
+        # get token IDs, shape: (*, num_chosen_tokens)
+        token_ids = self.assignment
+        if indices is not None:
+            token_ids = token_ids[indices]
+
+        # lookup token representations, shape: (*, num_chosen_tokens, *shape)
+        return self.vocabulary(token_ids)
+
+
+class NodePieceRepresentation(Representation):
+    r"""
+    Basic implementation of node piece decomposition [galkin2021]_.
+
+    .. math ::
+        x_e = agg(\{T[t] \mid t \in tokens(e) \})
+
+    where $T$ are token representations, $tokens$ selects a fixed number of $k$ tokens for each entity, and $agg$ is
+    an aggregation function, which aggregates the individual token representations to a single entity representation.
+
+    .. note ::
+        This implementation currently only supports representation of entities by bag-of-relations.
+    """
+
+    #: the token representations
+    token_representations: Sequence[TokenizationRepresentation]
+
+    def __init__(
+        self,
+        *,
+        triples_factory: CoreTriplesFactory,
+        token_representations: OneOrSequence[HintOrType[Representation]] = None,
+        token_representation_kwargs: OneOrSequence[OptionalKwargs] = None,
```

Comment:
```suggestion
        token_representations_kwargs: OneOrSequence[OptionalKwargs] = None,
```
we could add the changes from https://github.com/pykeen/pykeen/pull/715 here already

Reply:
great idea!

Reply:
Done in 0be29191